tests/testthat/test-reorder.R

```diff
@@ -69,11 +71,49 @@ test_that("complains if summary doesn't return single value", {
   })
 })
 
-test_that("fct_infreq respects missing values", {
-  f <- factor(c("a", "b", "b", NA, NA, NA), exclude = FALSE)
-  expect_equal(levels(fct_infreq(f)), c(NA, "b", "a"))
+
+# fct_infreq --------------------------------------------------------------
+
+test_that("fct_infreq() preserves explicit NA", {
+  x <- c("a", "b", "b", NA, NA, NA)
+  f <- factor(x, exclude = FALSE)
+  expect_equal(fct_infreq(f), fct(x, c(NA, "b", "a")))
+})
+
+test_that("fct_infreq() ignores implict NA", {
+  x <- c("a", "b", "b", NA, NA, NA)
+  f <- factor(x)
+  expect_equal(fct_infreq(f), fct(x, c("b", "a")))
+})
+
+test_that("fct_infreq() preserves empty levels", {
+  x <- c("b", "b", "a")
+  f <- fct(x, letters[1:3])
```

Comment:
Same as below, would be interesting to test the behavior when the empty is not already the last one.